### PR TITLE
Replace convert_syncbatchnorm in mmseg

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Remove `convert_syncbatchnorm` and use `mmcv.cnn.utils.revert_sync_batchnorm` since `revert_sync_batchnorm` is supported in mmcv by PR https://github.com/open-mmlab/mmcv/pull/1253. 

## Modification

- [x] Replace `convert_syncbatchnorm` with `revert_sync_batchnorm`
- [x] Remove `mmdeploy.codebase.mmseg.deploy.utils.convert_syncbatchnorm` and its unit test.
- [x] Update docs that [UNet](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/unet/fcn_unet_s5-d16_4x4_512x1024_160k_cityscapes.py) in mmseg only supports static shape due to concat failure in [UpConvBlock](https://github.com/open-mmlab/mmsegmentation/blob/7512f05990eb66bba3653cb4d5f478965bf41bd7/mmseg/models/utils/up_conv_block.py#L99)

## BC-breaking (Optional)

None

### CODE REVIEW

File: mmdeploy/codebase/mmseg/deploy/segmentation.py [hunk 1]
```diff
@@ -225,8 +225,7 @@ def evaluate_outputs(model_cfg,
                 to `False`.
         """
         if out:
-            logger = get_root_logger()
-            logger.info(f'\nwriting results to {out}')
+            logging.info(f'\nwriting results to {out}')
```
Comment (grimoire) [inline mmdeploy/codebase/mmseg/deploy/segmentation.py:228]: use logger in mmcv instead of logging.

**Comment by AllentDan** [comment]
Shall we rewrite the upsampling module to rescale the input `x` to the size of `skip`. This is actually what I did to mmocr

**Review by AllentDan — APPROVED** [review]

**Comment by AllentDan** [comment]
> Shall we rewrite the upsampling module to rescale the input `x` to the size of `skip`. This is actually what I did to mmocr

We can make `UNet` dynamically deployed in the future

**Comment by RunningLeon** [comment]
> Shall we rewrite the upsampling module to rescale the input `x` to the size of `skip`. This is actually what I did to mmocr
@AllentDan 
`Upsample` is set through config and there are many cases for it. Maybe should consider it in another pr if there has a request.

**Comment by RunningLeon** [comment]
Dynamic shape support with UNet will be done in mmseg repo. Related issue https://github.com/open-mmlab/mmsegmentation/issues/1236

**Review by grimoire — APPROVED** [review]

**Pull request merged**